### PR TITLE
`collectionStats` query fully implemented

### LINKED ISSUE #845: Pagination doesn't work when order by TOTAL_SALES
**Describe the bug**
When I set up a pagination ordering my products by TOTAL_SALES, the first 2 products constantly come back as first and second results.

**To Reproduce**
0 - Run the query below a first time and take note of the first 2 results
1 - Send the same query but this time with the cursor previously retrieved from the endCursor field
2 - Note that the first 2 results are the same as for the first query

```
query topSalesProducts($cursor: String, $first: Int) {
    products(
      where: {type: SIMPLE, orderby: {field: TOTAL_SALES, order: DESC}}
      first: $first
      after: $cursor
    ) {
      nodes {
        ... on SimpleProduct {
          databaseId
          name
        }
      }
      pageInfo {
        endCursor
      }
    }
  } 
```

**Expected behavior**
The first 2 results should not reappear once the cursor has been filled in

**Plugin Versions**
- **WooGraphQL Version:** 0.18.3
- **WPGraphQL Version:** 1.23.0
- **WordPress Version:** 6.5.2
- **WooCommerce Version:** 7.9.0

**Additional context**
The bug only appear since the v0.16.0

### PR DESCRIPTION
**Pull request by kidunot89**
### Your checklist for this pull request
Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.

🚨Please review the [guidelines for contributing](./CONTRIBUTING.md) to this repository.

- [x] Make sure you are making a pull request against the **develop branch** (left side). Also you should start *your branch* off *our develop*.
- [x] Make sure you are requesting to pull request from a **topic/feature/bugfix/devops branch** (right side). Don't pull request from your master!
- [x] Have you ensured/updated that CLI tests to extend coverage to any new logic. Learn how to modify the tests [here](https://woographql.com/contributing/2-local-testing).

What does this implement/fix? Explain your changes.
---------------------------------------------------
- Fixes regressions created by WPGraphQL v1.24.x
- Fixes `collectionStats` query parameters
- Adds `attributes` field to the `*ToProductUnionConnectionWhereArgs`  input types
- Deprecates `attribute` and `attributeTerm` on the  `*ToProductUnionConnectionWhereArgs` input types
- Requires `collectionStats` query and `attributes` field functionality introduced in WooCommerce v3.9.0

Does this close any currently open issues?
------------------------------------------
Resolves #845 

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
…

Where has this been tested?
---------------------------

- **WooGraphQL Version:** …
- **WPGraphQL Version:** …
- **WordPress Version:**
- **WooCommerce Version:**

### CODE REVIEW

File: includes/connection/class-products.php [hunk 1]
```diff
@@ -379,13 +379,19 @@ public static function get_connection_args( $extra_args = [] ): array {
...
 				'description' => __( 'Limit result set to products assigned a specific shipping class ID.', 'wp-graphql-woocommerce' ),
 			],
+			'attributes'         => [
+				'type'        => 'ProductAttributeQueryInput',
+				'description' => __( 'Limit result set to products with selected global attribute queries.', 'wp-graphql-woocommerce' ),
+			],
 			'attribute'          => [
-				'type'        => 'String',
-				'description' => __( 'Limit result set to products with a specific attribute. Use the taxonomy name/attribute slug.', 'wp-graphql-woocommerce' ),
+				'type'              => 'ProductAttributeEnum',
```
Comment (justlevine) [inline includes/connection/class-products.php:387]: Changing the type makes this breaking for schemas already using it. If it's deprecated anyway is there a need to change the type as well?

File: includes/data/connection/class-product-connection-resolver.php [hunk 2]
```diff
@@ -34,6 +33,13 @@ class Product_Connection_Resolver extends AbstractConnectionResolver {
 	 */
 	protected $post_type;
 
+	/**
+	 * The instance of the class that helps filtering with the product attributes lookup table.
+	 *
+	 * @var Filterer
```
Comment (justlevine) [inline includes/data/connection/class-product-connection-resolver.php:39]:
```suggestion
	 * @var \Automattic\WooCommerce\Internal\ProductAttributesLookup\Filterer
```

Add `|\stdClass` ,if this prop is overloadable.

**Review by justlevine — COMMENTED** [review]
Commented on the few things in the diff that jumped out at me, nothing pressing.

File: includes/connection/class-products.php (hunk 1 above)
Comment (kidunot89) [reply includes/connection/class-products.php:387]: No, there isn't thanks for letting me know. I'll change it back before merging

File: includes/data/connection/class-product-connection-resolver.php (hunk 2 above)
Comment (kidunot89) [reply includes/data/connection/class-product-connection-resolver.php:39]: It's not overloadable as fair as I know.

**Pull request merged**